latex/epoch.tex

```diff
@@ -1,54 +1,614 @@
-The pool reaping in rules are discribed in
-Figure~\ref{fig:ts-types:pool-reap} and Figure~\ref{fig:rules:pool-reap}.
+\newcommand{\UTxOEpEnv}{\type{UTxOEpEnv}}
+\newcommand{\UTxOEpState}{\type{UTxOEpState}}
+\newcommand{\Accnt}{\type{Accnt}}
+\newcommand{\AccntEnv}{\type{AccntEnv}}
+\newcommand{\AccntState}{\type{AccntState}}
+\newcommand{\StPlCleanEnv}{\type{StPlCleanEnv}}
+\newcommand{\StPlCleanState}{\type{StPlCleanState}}
+\newcommand{\NewProtoConstsEnv}{\type{NewProtoConstsEnv}}
+\newcommand{\NewProtoConstsState}{\type{NewProtoConstsState}}
+\newcommand{\EpochEnv}{\type{EpochEnv}}
+\newcommand{\EpochState}{\type{EpochState}}
+\newcommand{\Production}{\type{Production}}
+
+\newcommand{\obligation}[4]{\fun{obligation}~ \var{#1}~ \var{#2}~ \var{#3}~ \var{#4}}
+\newcommand{\reward}[5]{\fun{reward}~ \var{#1}~ \var{#2}~ \var{#3}~ \var{#4}~ \var{#5}}
+\newcommand{\poolRefunds}[3]{\fun{poolRefunds}~ \var{#1}~ \var{#2}~ \var{#3}}
 
 %%
-%% Figure - Pool Reap
+%% Figure - Epoch Abstract Types
 %%
+\begin{figure}
+  \emph{Abstract types}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \var{production} & \Production & \text{blocks produced in the epoch}\\
+    \end{array}
+  \end{equation*}
+  \caption{Epoch definitions}
+  \label{fig:epoch-defs}
+\end{figure}
+
+%%
+%% Figure - Functions for Epoch Rules
+%%
+\begin{figure}
+  \begin{align*}
+      & \fun{obligation} \in \PrtclConsts \to \Allocs \to \Allocs \to \Slot \to \Coin
+      & \text{total possible refunds} \\
+      & \obligation{pc}{stkeys}{stpools}{cslot} =\\
+      & \sum\limits_{(\_ \mapsto s) \in \var{stkeys}}
+        \refund{k_{\mathsf{val}}}{k_{\min}}{(\slotminus{cslot}{s})}{\lambda} \\
+      & + \sum\limits_{(\_ \mapsto s) \in \var{stpools}}
+        \refund{p_{\mathsf{val}}}{p_{\min}}{(\slotminus{cslot}{s})}{\lambda} \\
+      &
+        \where k_{\mathsf{val}}, k_{\min},p_{\mathsf{val}}, p_{\min}, \lambda \in pc
+      \nextdef
+      & \fun{reward} \in \Production \to \PrtclConsts \to \Coin \to DWState \to \\
+      & ~~~ (\AddrRWD \mapsto \Coin) \to (\AddrRWD \mapsto \Coin)
+      & \text{update rewards} \\
+      & \reward{prod}{pc}{availablePool}{dwstate}{rewards} = \mathsf{TBD}\\
+      \nextdef
+      & \fun{poolRefunds} \in \PrtclConsts \to \Allocs \to \Slot \to (\HashKey \mapsto \Coin)
+      & \text{pool refunds} \\
+      & \poolRefunds{pc}{retiring}{cslot} = \\
+      & \bigcup_{\var{hk}\mapsto s\in\var{retiring}}
+          \var{hk}\mapsto( \refund{p_{\mathsf{val}}}{p_{\min}}{(\slotminus{cslot}{s})}{\lambda} ) \\
+      &
+        \where p_{\mathsf{val}}, p_{\min}, \lambda \in pc
+  \end{align*}
+  \caption{Functions used in Deposits}
+  \label{fig:functions:epoch}
+\end{figure}
 
 
+%%
+%% Figure - UTxO Epoch Defs
+%%
 \begin{figure}
-  \emph{Pool Reap Transition}
+  \emph{UTxO Epoch environment}
+  \begin{equation*}
+    \UTxOEpEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
```

Comment:
having the 3rd column aligned to the right looks a bit strange

Reply:
ok, let's align to the left then!